src/test/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtFacilityServiceTest.java

```diff
@@ -0,0 +1,222 @@
+
+        when(court.getFacilities()).thenReturn(COURT_FACILITIES);
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+        when(courtRepository.save(court)).thenReturn(court);
+        when(facilityTypeRepository.findByName(FACILITY_TYPE_NAME)).thenReturn(Optional.of(facilityType));
+
+        assertThat(adminCourtFacilityService.addCourtFacility(COURT_SLUG,FACILITY))
+            .hasSize(4)
+            .containsExactlyElementsOf(expectedCourtFacilities);
+    }
+
+
+    @Test
+    void shouldDisplayMessageWhenFacilityAlreadyExists() {
+
+
+        when(court.getFacilities()).thenReturn(COURT_FACILITIES);
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+        when(courtRepository.save(court)).thenReturn(court);
+        when(facilityTypeRepository.findByName(FACILITY_TYPE_NAME)).thenReturn(Optional.of(facilityType));
+
+        assertThatThrownBy(() -> adminCourtFacilityService.addCourtFacility(COURT_SLUG, new uk.gov.hmcts.dts.fact.model.admin.Facility(FACILITY_3,DESCRIPTION_3,DESCRIPTION_CY_3)))
+            .isInstanceOf(DuplicatedListItemException.class)
+            .hasMessage(DUPLICATE_ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenAddingCourtFacilityForNonExistentCourt() {
+
+
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.empty());
+
+        assertThatThrownBy(() -> adminCourtFacilityService.addCourtFacility(COURT_SLUG,FACILITY))
+            .isInstanceOf(NotFoundException.class)
+            .hasMessage(NOT_FOUND + COURT_SLUG);
+    }
+
+    @Test
+    void shouldDeleteCourtFacility() {
+
+        final List<uk.gov.hmcts.dts.fact.model.admin.Facility> expectedCourtFacilities = Arrays.asList(
+            new uk.gov.hmcts.dts.fact.model.admin.Facility(FACILITY_1,DESCRIPTION_1,DESCRIPTION_CY_1),
+            new uk.gov.hmcts.dts.fact.model.admin.Facility(FACILITY_2,DESCRIPTION_2,DESCRIPTION_CY_2));
+
+        final uk.gov.hmcts.dts.fact.model.admin.Facility deletedFacility = new uk.gov.hmcts.dts.fact.model.admin.Facility(FACILITY_3,DESCRIPTION_3,DESCRIPTION_CY_3);
+
+        when(court.getFacilities()).thenReturn(COURT_FACILITIES);
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+        when(courtRepository.save(court)).thenReturn(court);
+        when(facilityTypeRepository.findByName(FACILITY_TYPE_NAME)).thenReturn(Optional.of(facilityType));
+
+        assertThat(adminCourtFacilityService.deleteCourtFacility(COURT_SLUG,deletedFacility))
+            .hasSize(2)
+            .containsExactlyElementsOf(expectedCourtFacilities);
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenDeletingCourtFacilityForNonExistentCourt() {
+
+
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.empty());
+
+        assertThatThrownBy(() -> adminCourtFacilityService.addCourtFacility(COURT_SLUG,FACILITY))
```

Comment:
The test method says it is deleting court facility but you are adding facility in the test. Copy ad paste error :-)

Reply:
replaced with delete method